### PR TITLE
fix:when skelton data destory clean all data.

### PR DESCRIPTION
**Pull request by zhakesi**
engine pr: [https://github.com/cocos-creator/engine/pull/9139](https://github.com/cocos-creator/engine/pull/9139)
previous version the info still exists in the data map,
this cause initializing skeleton failed second time.

[https://forum.cocos.org/t/topic/118416/11](https://forum.cocos.org/t/topic/118416/11)

### CODE REVIEW

File: cocos/editor-support/spine-creator-support/SkeletonDataMgr.cpp [hunk 1]
```diff
@@ -99,15 +98,12 @@ void SkeletonDataMgr::releaseByUUID(const std::string &uuid) {
...
-    if (info->getReferenceCount() == 1) {
-        _dataMap.erase(dataIt);
-        if (_destroyCallback) {
-            auto &texturesIndex = info->texturesIndex;
-            for (auto it = texturesIndex.begin(); it != texturesIndex.end(); it++) {
-                _destroyCallback(*it);
-            }
+    _dataMap.erase(dataIt);
+    if (_destroyCallback) {
+        auto &texturesIndex = info->texturesIndex;
```
Comment (mmyduckx) [inline cocos/editor-support/spine-creator-support/SkeletonDataMgr.cpp:103]: `auto &texturesIndex = info->texturesIndex;`
这行没用了吧

**Review by mmyduckx — CHANGES_REQUESTED** [review]

File: cocos/editor-support/spine-creator-support/SkeletonDataMgr.cpp (hunk 1 above)
Comment (zhakesi) [reply cocos/editor-support/spine-creator-support/SkeletonDataMgr.cpp]: 是，删了

**Review by mmyduckx — APPROVED** [review]

File: cocos/editor-support/spine-creator-support/SkeletonDataMgr.cpp [hunk 2]
```diff
@@ -89,7 +89,6 @@ SkeletonData *SkeletonDataMgr::retainByUUID(const std::string &uuid) {
     if (dataIt == _dataMap.end()) {
         return nullptr;
     }
-    dataIt->second->retain();
```
Comment (PatriceJiang) [inline cocos/editor-support/spine-creator-support/SkeletonDataMgr.cpp:92]: 如果不需要 retain,  那方法 `retainByUUID` 是不是要改成 `findByUUID`, `releaseByUUID` 改成 `deleteByUUID` ?

Comment (PatriceJiang) [reply cocos/editor-support/spine-creator-support/SkeletonDataMgr.cpp:92]: 感觉 Ref 机制本身没有问题, 应该是 retain/release 没匹配

**Review by PatriceJiang — APPROVED** [review]

**Review by mmyduckx — APPROVED** [review]

**Pull request merged**